sunpy/data/data_manager/manager.py

```diff
@@ -0,0 +1,145 @@
+import pathlib
+import functools
+from contextlib import contextmanager
+
+from sunpy.util.util import hash_file
+
+
+class DataManager:
+    """
+    This class provides a remote data manager for managing remote files.
+
+    Parameters
+    ----------
+    cache: `sunpy.data.data_manager.cache.Cache`
+        Cache object to be used by `~sunpy.data.data_manager.manager.DataManager`.
+    """
+
+    def __init__(self, cache):
+        self._cache = cache
+
+        self._file_cache = {}
+
+        self._skip_hash_check = False
+        self._skip_file = {}  # Dict[str, str]
```

Comment:
I am not sure if I am on board with using a comment because we could just use type hints, but then I am not sure I want to unleash that particular genie either, so maybe I am just ok with this!!